services/ui-src/src/routes/Dashboard/Dashboard.tsx

```diff
@@ -89,6 +92,30 @@ export const Dashboard = () => {
   const openAddEditProgramModal = (reportId?: string) => {
     // if reportId provided, set as selected program
     setSelectedReportId(reportId);
+
+    const selectedReport = reportsByState?.find(
+      (o: { reportId: string }) => o.reportId === reportId
+    );
```

Comment:
If we pass in the whole report metadata object to this function instead of passing just the reportId, we can save ourselves this `.find` and make the checks below more clear too.